doc/syscall-limitations.md

```diff
@@ -0,0 +1,142 @@
+| SYS_rt_sigqueueinfo / SYS_rt_tgsigqueueinfo | deliver a signal with siginfo | Unhanlded |
+| SYS_rt_sigsuspend              | replace the signal mask and wait for a signal | Unsupported |
+| SYS_sigaltstack                | Install an alternative stack for certain signal handlings | Stub only |
+
+## System calls related to file or file system operations
+
+| Syscall names        | Description             | Compatibility |
+| -------------------- |-------------------| --------------|
+| SYS_fcntl | File descriptor operations | Partial |
+| SYS_mknod | Create a file system node  | Partial |
+| SYS_openat / SYS_mkdirat / SYS_mknodat / SYS_fchownat / SYS_futimesat / SYS_unlinkat / SYS_renameat / SYS_renameat2 / SYS_linkat / SYS_symlinkat / SYS_readlinkat / SYS_fchmodat / SYS_faccessat | File system operation relative to a directory file descriptor | Unsupported |
+| SYS_lsetxattr / SYS_fsetxattr / SYS_getxattr / SYS_lgetxattr / SYS_fgetxattr / SYS_listxattr / SYS_llistxattr / SYS_flistxattr / SYS_removexattr / SYS_lremovexattr/ SYS_fremovexattr | get/set/remove extended file attributes | Unsupported |
+| SYS_inotify_add_watch | Monitor file system changes | Partial |
+| SYS_fanotify_init / SYS_fanotify_mark | Monitor file system changes | Unsupported |
+| SYS_statx | get extended file status | Unsupported |
+| SYS_memfd_create | create anonymous file descriptor | Unsupported |
+| SYS_chroot | change root directory | Unsupported |
+| SYS_statfs / SYS_fstatfs | get file system status | Partial |
+| SYS_sync / SYS_syncfs | commit filesystem caches to disk | Unsupported |
+
+## System calls related to low level memory management
+
+| Syscall names        | Description             | Compatibility |
+| -------------------- |-------------------| --------------|
+| SYS_mmap                | allocate memory pages | Partial |
+| SYS_msync               | flush mmaped files | Partial |
+| SYS_mlock               | lock pages in memory | Stub only |
+| SYS_mlock2 / SYS_munlock / SYS_mlockall / SYS_munlockall | lock/unlock pages in memory | Unsupported |
+| SYS_pkey_mprotect / SYS_pkey_alloc / SYS_pkey_free | MPK based page protection | Unsupported |
+| SYS_mincore             | get memory residency status of pages | Unsupported |
+| SYS_mbind / SYS_set_mempolicy / SYS_get_mempolicy | get/set NUMA policy for memory pages | Unsupported |
+| SYS_migrate_pages / SYS_move_pages | Move memory pages across nodes | Unsupported |
+
+
+## Miscellaneous system calls
+
+| Syscall names        | Description             | Compatibility |
+| -------------------- |-------------------| --------------|
+| SYS_syslog              | read and/or clear kernel message ring buffer | Unsupported |
+| SYS_futex               | futex operations | Partial |
+| SYS_get_thread_area     | get thread-local storage | Unsupported |
+| SYS_clock_nanosleep     | high-resolution sleep | Unsupported |
```

Comment:
It seems that nanosleep() is supported through an OCALL

Reply:
That's true for `SYS_nanosleep`. `clock_nanosleep` is more like an advanced `nanosleep` in which you can choose which clock to sleep on.